src/box.ts

```diff
@@ -0,0 +1,33 @@
+export interface BoxInterface {
+  x: number
+  y: number
+  width: number
+  height: number
+  readonly horizontals: number[]
+  readonly verticals: number[]
+}
+export default class Box {
+  x: number
+  y: number
+  width: number
+  height: number
+
+  constructor(options: BoxInterface) {
```

Comment:
Don't think we need an interface here. Typed arguments will be enough i.e. `options: { x: number, etc: ... }`.